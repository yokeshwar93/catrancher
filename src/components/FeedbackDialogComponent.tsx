import React from "react";

import DialogTitle from '@material-ui/core/DialogTitle';
import Dialog from '@material-ui/core/Dialog';
import {Button, makeStyles, Typography, withStyles} from "@material-ui/core";
import {DialogProps} from "../models/modelTypes";

const StyledDialog = withStyles({
    paper: {
        width: '50vw'
    }
})(Dialog);




const useStyles = makeStyles((theme) => ({
    title: {
        fontWeight: 'bold',
        fontSize: '2em'
    },
    message: {
        marginLeft: '4%',
        marginBottom: '5%'
    },

    container: {
        width: '20vw',
    },
    button: {
        borderRadius: '0px'
    },

}));

const FeedbackDialogComponent = (props : DialogProps) => {

    const classes = useStyles();

    const { openDialog, feedback, handleClose } = props;

    return (
        <StyledDialog onClose={handleClose} aria-labelledby="simple-dialog-title" open={openDialog}>
            <DialogTitle className={`${classes.title} ${feedback.titleClass}`}>{feedback.titleText}</DialogTitle>
            <Typography className={`${classes.message} ${feedback.messageClass}`}>{feedback.messageText}</Typography>
            <Button onClick={handleClose} className={classes.button} variant={'contained'} color={feedback.isSuccess ? 'primary' : 'secondary'}>{feedback.buttonText}</Button>
        </StyledDialog>
    )
}

export default FeedbackDialogComponent;


