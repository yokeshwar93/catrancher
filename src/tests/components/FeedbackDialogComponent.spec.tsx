import React from 'react';
import { render } from '@testing-library/react';
import { mockFeedbackValue } from "../store/reducer/mockData";
import FeedbackDialogComponent from "../../components/FeedbackDialogComponent";

describe('renders feedback dialog', () => {

    it('render feedback dialog component', () => {
        const { container } = render(
            <FeedbackDialogComponent feedback={mockFeedbackValue} openDialog={true} handleClose={jest.fn()}  />
        );
        expect(container).toMatchSnapshot();

    })

});
