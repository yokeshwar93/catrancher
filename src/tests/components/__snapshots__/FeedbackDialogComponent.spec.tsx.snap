// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders feedback dialog render feedback dialog component 1`] = `
<div
  aria-hidden="true"
/>
`;
